src/main/java/com/google/cloud/genomics/dataflow/pipelines/LoadReadsToBigQuery.java

```diff
@@ -0,0 +1,377 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.genomics.dataflow.pipelines;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.common.collect.Lists;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.util.gcsfs.GcsPath;
+import org.apache.beam.sdk.values.PCollection;
+import com.google.cloud.genomics.dataflow.readers.bam.BAMShard;
+import com.google.cloud.genomics.dataflow.readers.bam.ReadBAMTransform;
+import com.google.cloud.genomics.dataflow.readers.bam.Reader;
+import com.google.cloud.genomics.dataflow.readers.bam.ReaderOptions;
+import com.google.cloud.genomics.dataflow.readers.bam.ShardingPolicy;
+import com.google.cloud.genomics.dataflow.utils.GCSOptions;
+import com.google.cloud.genomics.dataflow.utils.GCSTempPathOptions;
+import com.google.cloud.genomics.dataflow.utils.GenomicsOptions;
+import com.google.cloud.genomics.dataflow.utils.ShardOptions;
+import com.google.cloud.genomics.utils.Contig;
+import com.google.cloud.genomics.utils.OfflineAuth;
+import com.google.cloud.genomics.utils.ShardBoundary;
+import com.google.common.base.Strings;
+import com.google.genomics.v1.Read;
+import com.google.genomics.v1.CigarUnit;
+
+import htsjdk.samtools.ValidationStringency;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Pipeline for loading BAM file (hosted on GCS) into BigQuery.
+ *
+ */
+public class LoadReadsToBigQuery {
+
+  public static interface Options extends GCSOptions, ShardOptions, GCSTempPathOptions {
+
+    @Description("The Google Cloud Storage path to the BAM file to get reads data from.")
+    @Default.String("")
+    String getBAMFilePath();
+    void setBAMFilePath(String filePath);
+
+    @Description("Whether to shard BAM file reading.")
+    @Default.Boolean(true)
+    boolean isShardBAMReading();
+    void setShardBAMReading(boolean newValue);
+
+    @Description("For shareded BAM file input, the maximum size in bytes of a shard processed by "
```

Comment:
(1) Can we just have "shard_size_mb" and if it's non-zero we shard (eg drop other sharding flag)
(2) Is bytes really correct?  Presumably MB is the smallest reasonable unit...
(3) Why would the user *not* want to shard?  It would be best if we could make these decisions for the user (eg, based on the size of the input file for example)

Reply:
This pipeline mimics `CountReads.java` pipeline and the choices made there. Happy to change those that don't make sense, but the caveat is that I am not aware of historical reasons behind them. :)

(1) Done
(2) I think it's made in byte because the underlying Interface (i.e. `shard.approximateSizeInBytes`) is in byte. I changed this flag to MB. 
(3) Not sure about the historical reasons. I can guess it would be useful to test if sharding logic works fine (i.e. output BQ table is the same with/without sharding).